controllers/druid/handler.go

```diff
@@ -1605,30 +1599,3 @@ func getAllNodeSpecsInDruidPrescribedOrder(m *v1alpha1.Druid) ([]keyAndNodeSpec,
 func namespacedName(name, namespace string) *types.NamespacedName {
 	return &types.NamespacedName{Name: name, Namespace: namespace}
 }
-
-//-------------------------------------------
-// resetGroupVersionKind func is copied from controller-runtime/pkg/client/client.go to retain TypeMeta
-// on sdk.Create/Delete , PATCH/UPDATE already retain that
-
-// resetGroupVersionKind is a helper function to restore and preserve GroupVersionKind on an object.
-// TODO(vincepri): Remove this function and its calls once    controller-runtime dependencies are upgraded to 1.15.
-func resetGroupVersionKind(obj runtime.Object, gvk schema.GroupVersionKind) {
-	if gvk != schema.EmptyObjectKind.GroupVersionKind() {
-		if v, ok := obj.(schema.ObjectKind); ok {
-			v.SetGroupVersionKind(gvk)
-		}
-	}
-}
-
-// Create implements client.Client
-func sdkCreate(ctx context.Context, sdk client.Client, obj runtime.Object) error {
-	defer resetGroupVersionKind(obj, obj.GetObjectKind().GroupVersionKind())
-	return sdk.Create(ctx, obj)
-}
-
-func sdkDelete(ctx context.Context, sdk client.Client, obj runtime.Object) error {
-	defer resetGroupVersionKind(obj, obj.GetObjectKind().GroupVersionKind())
-	return sdk.Delete(ctx, obj)
-}
-
```

Comment:
@himanshug is it fine to remove these functions since i dont see they being called in the handler. 
Regardless with the interface not sure if they are of any use. Let me know :)

Reply:
sure, if they aren't used anywhere.